collection-spec/collection-spec.md

```diff
@@ -24,7 +24,7 @@ Implementations are encouraged, however, as good effort will be made to not chan
 | Element      | Type              | Description                                                  |
 | ------------ | ----------------- | ------------------------------------------------------------ |
 | stac_version | string            | **REQUIRED.** The STAC version the collection implements.    |
-| id           | string            | **REQUIRED.** Identifier for the collection that is unique across the provider. The ``/`` character is forbidden since the id value is used in some STAC API endpoints. |
+| id           | string            | **REQUIRED.** Identifier for the collection that is unique across the provider. Must be [percented encoded](https://tools.ietf.org/html/rfc3986#section-2) if a reserved character is used. |
 | title        | string            | A short descriptive one-line title for the collection.       |
```

Comment:
This may be removed if the percent encoding is mentioned in api-spec.

Reply:
We should put this somewhere differently. Percent encoding the ID in JSON doesn't make any sense. Before making any action, let's better figure out in which scenario this makes problems. 

For sure, you need to encode it in the generated links. So putting a description or link there would make sense. Similarly it could be added to the API. Are there other places where the links are not taken from the links section, but generated by clients / manually (i.e. not by the generating servers in the links section)?

Reply:
Yes, I kept it there for discussion because that's what I understood @matthewhanson suggested on our last call. The PR currently has it also mentioned in the api-spec definition and one of them should go depending on the final decision.

I agree that mentioning it in the API spec is cleaner if we not restrict the allowed charset for ids. The only advantage I see in percent encoding the id is to warn the static stac implementer that some ids may generate awkward endpoints for dynamic APIs built over its content. 

That was my case, I would not choose CBERS4/MUX in the first place if I was aware of that point,
CBERS4-MUX works just fine.

Maybe we could just give a heads-up in the IDs definition that some chars would be escaped in some API endpoints.